lib/puppet/functions/regsubst.rb

```diff
@@ -95,7 +95,27 @@ def regsubst_regexp(target, pattern, replacement, flags = nil)
   end
 
   def inner_regsubst(target, re, replacement, op)
-    target.respond_to?(op) ? target.send(op, re, replacement) : target.collect { |e| e.send(op, re, replacement) }
+    if target.respond_to?(:map)
```

Comment:
I'm not a big fan of checking if something is an `Array`/`Sensitive` by checking if it responds to `map`/`unwrap`. For example, all classes that include `Enumerable` will respond to `map`. I usually prefer checking if the object is an instance of the expected class or a subclass, e.g. `item.is_a?(Array)`

Reply:
I'd like to argue opposite-wise:  If I want the Duck to quack, then I have to test, if she is able to quack.  
If you test for a particular Datatype, then you have to cover all possible Cases.  
Not implemented.

Reply:
Have changed it…